airbyte-commons-worker/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java

```diff
@@ -107,6 +108,8 @@ public Process create(
 
       final var allLabels = getLabels(jobId, attempt, customLabels);
 
+      final var nodeSelectors = usesIsolatedPool ? workerConfigs.getWorkerIsolatedKubeNodeSelectors() : workerConfigs.getworkerKubeNodeSelectors();
```

Comment:
it can be empty if the flag was set to false - if they are using custom connectors but didn't turn on the useIsolatedNodePool flag, they might hit this